src/query/service/src/pipelines/processors/transforms/hash_join/transform_hash_join_probe.rs

```diff
@@ -203,6 +203,21 @@ impl TransformHashJoinProbe {
             {
                 self.join_probe_state.probe_done()?;
                 Ok(Event::Async)
+            } else if self
+                .join_probe_state
+                .hash_join_state
+                .merge_into_need_target_partial_modified_scan()
```

Comment:
Is there a situation where both `need_outer_scan` and `merge_into_need_target_partial_modified_scan` are true?

Reply:
No, this optimization is just for left join for now, I will add Inner, but all of these is just for the situation where the target table as build side.(MergeInto's default join is right join,in this case the source data as build side).